sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +260,141 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._configuration_refresh = self._ConfigurationRefresh(**kwargs)
+
+    def refresh_configuration_settings(self, func):
+        @wraps(func)
+        def refresh_wrapper(*args, **kwargs):
+            self.refresh()
+            return func(*args, **kwargs)
+
+        return refresh_wrapper
+
+    def refresh(self, **kwargs) -> None:
+        # pylint:disable=protected-access
+        refresh_registrations = self._configuration_refresh.refresh_options._refresh_registrations
+        if len(refresh_registrations) == 0:
+            logging.debug("Refresh called but no refresh options set.")
+            self._configuration_refresh.refresh_options._on_error()
+            return
+
+        if datetime.now() < self._configuration_refresh.next_refresh_time:
+            logging.debug("Refresh called but refresh interval not elapsed.")
```

Comment:
This could be a lot of logs if refresh is called in the application hot path

Reply:
That is why it is a debug log, not info or warning. It should never be on in prod.

Reply:
It could still cause bloated debug logs. I am not sure where users would call refresh in Python apps, but in .NET hot path, we may see 10,000 requests per second. That's a huge amount of logs for any level. 
Even if we skip this statement, I dont think we're missing much. We should be adding logs when refresh is actually triggered - i.e. when refresh interval has elapsed.

Reply:
I don't really mind either way - though as @mrm9084 says, customers should never turn on debug logs in prod. So if they _do_ get bloated logs, this line wont be the only reason....